Scratch mode generates code with missing dependencies

## Summary

When the AI template selector returns no match and the system falls back to scratch mode, generated applications consistently fail at runtime due to missing npm dependencies. The LLM generates code referencing packages (e.g., `tailwindcss-animate`, `@radix-ui/react-slot`) that are never installed.

## Root Cause

The issue stems from the interaction between three components:

1. **Empty scratch template** - `createScratchTemplateDetails()` (`worker/agents/utils/templates.ts:133-148`) returns `allFiles: {}`, `deps: {}`, `frameworks: []`
2. **Locked package.json** - The system prompt (`worker/agents/prompts.ts:841`) instructs the LLM: "Refrain from editing 'dont touch' files: package.json, vite.config.ts, wrangler.jsonc"
3. **No dependency installation mechanism** - There is no step in the scratch mode flow for the LLM to request new packages be added

The LLM generates shadcn/ui and Radix UI code from its training data, but the generated `package.json` only includes what the empty scratch template provides. The dev server inside the sandbox container crashes on first import of a missing package.

## Code Path

```
User prompt (no template match)
  → handleAITemplateSelection() returns null (worker/agents/index.ts:155-158)
  → handleGeneralType() (worker/agents/index.ts:86-97)
  → createScratchTemplateDetails() (worker/agents/utils/templates.ts:133-148)
  → Empty deps + LLM told package.json is locked
  → LLM generates code referencing uninstalled packages
  → Container dev server crashes: "Cannot find module 'tailwindcss-animate'"
```

## Reproduction

1. Start a new chat session
2. Use a prompt that doesn't match any template in the catalog (e.g., a generic request that the AI selector scores below threshold)
3. The system falls back to scratch mode
4. Generated app fails at runtime with missing module errors

## Observed Symptoms

- Container dev server crash: `Fatal error detected: Cannot find module 'tailwindcss-animate'`
- Preview iframe enters retry loop (HEAD requests return 502 from tunnel / connection refused from sandbox)
- After 10 retries, user sees "Preview Not Available" with no indication the cause is missing dependencies
- Generated `package.json` uses React 19 with minimal deps, contrasting with template-based apps that include 50+ dependencies

## Possible Approaches

Not prescribing a solution, but some directions that might be worth considering:

- Allow the LLM to specify dependencies during scratch mode generation (relax the package.json lock for this flow)
- Pre-populate the scratch template with a baseline set of common dependencies
- Add a post-generation step that analyzes imports and runs `bun add` for missing packages
- Surface container crash logs to the code fixer/debugger so it can diagnose and install missing deps

--- a/worker/agents/utils/templates.ts
+++ b/worker/agents/utils/templates.ts
@@ -37,7 +37,6 @@ const emitLog = (level: "info" | "warn" | "error", rawMessage: string) => {
   }
 };
 
-// 3. Create the custom logger for Vite
 const customLogger = {
   warnOnce: (msg: string) => emitLog("warn", msg),
 
@@ -97,33 +96,74 @@ export default ({ mode }: { mode: string }) => {
 
 `;
 
+const SCRATCH_PACKAGE_JSON = `{
+  "name": "scratch-project",
+  "private": true,
+  "version": "0.0.0",
+  "type": "module",
+  "scripts": {
+    "dev": "vite --host 0.0.0.0 --port \${PORT:-8001}",
+    "build": "vite build",
+    "lint": "eslint --cache -f json --quiet .",
+    "preview": "bun run build && vite preview --host 0.0.0.0 --port \${PORT:-8001}",
+    "deploy": "bun run build && wrangler deploy",
+    "cf-typegen": "wrangler types"
+  },
+  "dependencies": {
+    "class-variance-authority": "^0.7.1",
+    "clsx": "^2.1.1",
+    "hono": "^4.8.5",
+    "lucide-react": "^0.525.0",
+    "pino": "^9.11.0",
+    "react": "^18.3.1",
+    "react-dom": "^18.3.1",
+    "react-router-dom": "6.30.0",
+    "tailwind-merge": "^3.3.1",
+    "tailwindcss-animate": "^1.0.7"
+  },
+  "devDependencies": {
+    "@cloudflare/vite-plugin": "^1.9.4",
+    "@cloudflare/workers-types": "^4.20250807.0",
+    "@types/node": "^22.15.3",
+    "@types/react": "^18.3.1",
+    "@types/react-dom": "^18.3.1",
+    "@vitejs/plugin-react": "^4.3.4",
+    "autoprefixer": "^10.4.21",
+    "postcss": "^8.5.3",
+    "tailwindcss": "^3.4.17",
+    "typescript": "5.8",
+    "vite": "^6.3.1"
+  }
+}
+`;
+
+const SCRATCH_WRANGLER_JSONC = `{
+  "name": "scratch-project",
+  "main": "worker/index.ts",
+  "compatibility_date": "2025-04-01",
+  "compatibility_flags": ["nodejs_compat"],
+  "assets": {
+    "directory": "dist",
+    "not_found_handling": "single-page-application",
+    "run_worker_first": true,
+    "binding": "ASSETS"
+  },
+  "observability": {
+    "enabled": true
+  }
+}
+`;
+
 const SCRATCH_TEMPLATE_INSTRUCTIONS = `
 To build a valid, previewable and deployable project, it is essential to follow few important rules:
 
-1. The package.json **MUST** be of the following form: 
-\`\`\`
-...
-	"scripts": {
-		"dev": "vite --host 0.0.0.0 --port \${PORT:-8001}",
-		"build": "vite build",
-		"lint": "eslint --cache -f json --quiet .",
-		"preview": "bun run build && vite preview --host 0.0.0.0 --port \${PORT:-8001}",
-		"deploy": "bun run build && wrangler deploy",
-		"cf-typegen": "wrangler types"
-	}
-...
-\`\`\`
-
-Failure to have a compatible package.json would result in the app un-previewable and un-deployable.
+1. A baseline \`package.json\` is already provided with required scripts and core dependencies. You may modify it to add additional dependencies your code needs.
 
 2. The project **MUST** be a valid Cloudflare worker/durable object + Vite + bun project. 
 
-3. It must have a valid wrangler.jsonc and a vite.config.ts file.
+3. A \`wrangler.jsonc\` is already provided and **MUST NOT be modified**. It is preconfigured for the deployment environment.
 
-4. The vite config file MUST have the following minimal config:
-\`\`\`ts
-${VITE_CONFIG_MINIMAL}
-\`\`\`
+4. A \`vite.config.ts\` is already provided. **DO NOT modify it** unless absolutely necessary.
 `;
 
 /**
@@ -135,13 +175,17 @@ export function createScratchTemplateDetails(): TemplateDetails {
         name: 'scratch',
         description: { selection: 'from-scratch baseline', usage: `No template. Agent will scaffold as needed. **IT IS RECOMMENDED THAT YOU CHOOSE A VALID PRECONFIGURED TEMPLATE IF POSSIBLE** ${SCRATCH_TEMPLATE_INSTRUCTIONS}` },
         fileTree: { path: '/', type: 'directory', children: [] },
-        allFiles: {},
+        allFiles: {
+            'package.json': SCRATCH_PACKAGE_JSON,
+            'vite.config.ts': VITE_CONFIG_MINIMAL,
+            'wrangler.jsonc': SCRATCH_WRANGLER_JSONC,
+        },
         language: 'typescript',
         deps: {},
         projectType: 'general',
         frameworks: [],
         importantFiles: [],
-        dontTouchFiles: [],
+        dontTouchFiles: ['wrangler.jsonc'],
         redactedFiles: [],
         disabled: false,
     };

